v3/docs/03-runtime/k-testing/index.mdx

```diff
@@ -190,6 +190,25 @@ fn my_runtime_test() {
 }
 ```
 
+### Testing Events
+
+It can be useful to test that the emitted events conforms to some expectations.
+
+We can acess events emitted from `frame_system` pallet.
+
+```rust
+#[test]
+fn subtract_extrinsic_works() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(<test_pallet::Pallet<Test>>::subtract(Origin::signed(1), 42, 12));
+		let event = <frame_system::Pallet<Test>>::events().pop()
+			.expect("Expected Result EventRecord").event;
```

Comment:
Here we assume `EventRecord` is the event we're testing, right?

Reply:
From what I found in the docs `<frame_system::Pallet<Test>>::events()` returns a list of [`EventRecord`](https://docs.substrate.io/rustdocs/latest/frame_system/struct.EventRecord.html).

`Result` event is the event we are matching against.

Maybe the `Result` event or the expect error message needs to be clearer.

Reply:
Perhaps clearer to update the error message to:
`.expect("Expected event to match `EventRecord`")`

Reply:
What about "Expected at least one EventRecord to be found"